contracts/integrations/test/framework/assertions/finalizePool.ts

```diff
@@ -0,0 +1,269 @@
+    return new BigNumber(
+        feeRatio
+            .pow(COBB_DOUGLAS_ALPHA)
+            .times(stakeRatio.pow(toDecimal(1).minus(COBB_DOUGLAS_ALPHA)))
+            .times(toDecimal(rewardsAvailable))
+            .toFixed(0, BigNumber.ROUND_FLOOR),
+    );
+}
+
+interface FinalizePoolBeforeInfo {
+    poolStats: PoolStats;
+    aggregatedStats: AggregatedStats;
+    poolRewards: BigNumber;
+    cumulativeRewardsLastStored: BigNumber;
+    mostRecentCumulativeRewards: {
+        numerator: BigNumber;
+        denominator: BigNumber;
+    };
+}
+
+/**
+ * Returns a FunctionAssertion for `finalizePool` which assumes valid input is provided. The `after`
+ * callback below is annotated with the solidity source of `finalizePool`.
+ */
+/* tslint:disable:no-unnecessary-type-assertion */
+export function validFinalizePoolAssertion(
+    deployment: DeploymentManager,
+    simulationEnvironment: SimulationEnvironment,
+): FunctionAssertion<[string], FinalizePoolBeforeInfo, void> {
+    const { stakingWrapper } = deployment.staking;
+
+    return new FunctionAssertion<[string], FinalizePoolBeforeInfo, void>(stakingWrapper, 'finalizePool', {
+        before: async (args: [string]) => {
+            const [poolId] = args;
+            const { currentEpoch } = simulationEnvironment;
+            const prevEpoch = currentEpoch.minus(1);
+
+            const poolStats = PoolStats.fromArray(await stakingWrapper.poolStatsByEpoch(poolId, prevEpoch).callAsync());
+            const aggregatedStats = AggregatedStats.fromArray(
+                await stakingWrapper.aggregatedStatsByEpoch(prevEpoch).callAsync(),
+            );
+            const poolRewards = await stakingWrapper.rewardsByPoolId(poolId).callAsync();
+            const [
+                mostRecentCumulativeRewards,
+                cumulativeRewardsLastStored,
+            ] = await stakingWrapper.getMostRecentCumulativeReward(poolId).callAsync();
+            return {
+                poolStats,
+                aggregatedStats,
+                poolRewards,
+                cumulativeRewardsLastStored,
+                mostRecentCumulativeRewards,
+            };
+        },
+        after: async (beforeInfo: FinalizePoolBeforeInfo, result: FunctionResult, args: [string]) => {
+            // Ensure that the tx succeeded.
+            expect(result.success, `Error: ${result.data}`).to.be.true();
+
+            const logs = result.receipt!.logs; // tslint:disable-line:no-non-null-assertion
+
+            // // Compute relevant epochs
+            // uint256 currentEpoch_ = currentEpoch;
+            // uint256 prevEpoch = currentEpoch_.safeSub(1);
+            const { stakingPools, currentEpoch } = simulationEnvironment;
+            const prevEpoch = currentEpoch.minus(1);
+            const [poolId] = args;
+            const pool = stakingPools[poolId];
+
+            // // Load the aggregated stats into memory; noop if no pools to finalize.
```

Comment:
Should this be removed? Or if you wanted to keep the Solidity then could be good to add a note for future devs.

Reply:
eh yeah looking at this again I think I'll just replace these with normal comments